packages/webview_flutter/android/src/main/java/io/flutter/plugins/webviewflutter/FlutterWebView.java

```diff
@@ -29,6 +34,48 @@
   private final FlutterWebViewClient flutterWebViewClient;
   private final Handler platformThreadHandler;
 
+  // Verifies that a url opened by `Window.open` has a secure url.
+  private class FlutterWebChromeClient extends WebChromeClient {
+    @Override
+    public boolean onCreateWindow(
+        final WebView view, boolean isDialog, boolean isUserGesture, Message resultMsg) {
+      final WebViewClient webViewClient =
+          new WebViewClient() {
+            @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+            @Override
+            public boolean shouldOverrideUrlLoading(
+                @NonNull WebView view, @NonNull WebResourceRequest request) {
+              final String url = request.getUrl().toString();
+              if (isSecure(url)) {
+                flutterWebViewClient.shouldOverrideUrlLoading(FlutterWebView.this.webView, request);
+              }
+              return true;
+            }
+
+            @Override
+            public boolean shouldOverrideUrlLoading(WebView view, String url) {
+              if (isSecure(url)) {
+                flutterWebViewClient.shouldOverrideUrlLoading(FlutterWebView.this.webView, url);
```

Comment:
Does invoking `shouldOVerrideUrlLoading` has a side effect of loading the URL? It seems like our implementation of `shouldOVerrideUrlLoading` introduces such a side effect on older webview versions only if a navigation delegate is set. Does this work when no navigation delegate is set?

Reply:
I added a test, so it does still work without a navigation delegate.